packages/payload/src/config/types.ts

```diff
@@ -176,29 +177,12 @@ export type OGImageConfig = {
   width?: number | string
 }
 
-export type OpenGraphConfig = {
-  description?: string
-  images?: OGImageConfig | OGImageConfig[]
-  siteName?: string
-  title?: string
-}
-
-export type IconConfig = {
-  color?: string
-  /**
-   * @see https://developer.mozilla.org/docs/Web/API/HTMLImageElement/fetchPriority
-   */
-  fetchPriority?: 'auto' | 'high' | 'low'
-  media?: string
-  /** defaults to rel="icon" */
-  rel?: string
-  sizes?: string
-  type?: string
-  /**
-   * URL of the icon to use. You can use a relative path from the public folder (see https://nextjs.org/docs/app/building-your-application/optimizing/static-assets) or an absolute URL.
-   */
-  url: string
-}
+/**
+ * @todo find a way to remove the type assertion here.
+ * It can probably be removed after the `DeepRequired` from `Config` to `SanitizedConfig` is removed.
+ * Same with `CollectionConfig` to `SanitizedCollectionConfig`.
+ */
+type DeepClone<T> = T extends object ? { [K in keyof T]: DeepClone<T[K]> } : T
```

Comment:
I'm not understanding why DeepClone is used or needed here

Reply:
This is the result of a long hard battle with our use of `DeepRequired`. It doesn't seem to work as expected within a a complex intersection type. I.e. we import `Metadata` from `next` which contains many unions, etc. Cloning the type made it able to be fully traversed.